tensorflow_datasets/core/utils/gcs_utils.py

```diff
@@ -58,28 +55,23 @@ def gcs_path(suffix: Optional[str] = None) -> str:
 
 def download_gcs_file(path, out_fname=None, prefix_filter=None):
   """Download a file from GCS, optionally to a file."""
-  url = posixpath.join(GCS_BUCKET, path)
+  url = posixpath.join(GCS_ROOT_DIR, path)
   if prefix_filter:
-    url += '?prefix=%s' % prefix_filter
-  stream = bool(out_fname)
-  resp = requests.get(url, stream=stream)
-  if not resp.ok:
-    raise ValueError('GCS bucket inaccessible')
+    url = posixpath.join(url, prefix_filter)
   if out_fname:
-    with tf.io.gfile.GFile(out_fname, 'wb') as f:
-      for chunk in resp.iter_content(1024):
-        f.write(chunk)
+    dirpath, file = posixpath.split(out_fname)
+    if not tf.io.gfile.exists(dirpath):
+      tf.io.gfile.makedirs(dirpath)
```

Comment:
Why was `makedirs` required ? It wasn't required before ?

Reply:
Yes, It was not there before. I added to solve the error on my local machine and thought it will also be required here. I'm not sure it is necessary or not.

Reply:
In which case did you encounter the error. Do you remember ?

Reply:
When any file came for downloading(`out_fname=somepath`), If our local directory is not there then I got 
`NotFoundError: mnist/1.2.0/dataset_info.json; No such file or directory` where `mnist/1.2.0/` is my `local directory` path
I think we don't need this
```python
    dirpath, file = posixpath.split(out_fname)
    if not tf.io.gfile.exists(dirpath):
      tf.io.gfile.makedirs(dirpath)
```
Because, As per my understanding the directories are prebuilt before coming into `download_gcs_file` function
So we don't need to make directories again. Correct me if I'm wrong.
Also, I don't know where to test `gcs_utils.py`, So I test each function by passing some random datasets and local dirs from GCP

Reply:
Thanks I've imported this internally and will merge this soon.